addons/web/models/ir_http.py

```diff
@@ -25,6 +44,18 @@ def is_a_bot(cls):
         # timeit has been done to check the optimum method
         return any(bot in user_agent for bot in cls.bots)
 
+    @classmethod
+    def _pre_dispatch(cls, rule, args):
+        super()._pre_dispatch(rule, args)
+        debug = request.httprequest.args.get('debug', '')
+        if debug:
```

Comment:
`debug=` disable the debug mode

I always be surprised to see that ?debug enable de debug mode in the past (since v8 at least) because as you will reply, you consider than it is not set and so to ignore...

But like for oldies feature from odoo we try to not break the classic flow.

So, in case your are in debug mode, ?debug should remove it as previously.
For now, You cannot remove debug mode without change uri yourself to set debug=0.
```suggestion
        if 'debug' in request.httprequest.args:
```
```suggestion
        if 'debug' in request.params:
```
```suggestion
        if 'debug' in request.get_http_params():
```
I'm pretty sure you will use the best one ;)

It was a requirement to keep debug='' as a valid way because remove completely params is not valid (saved in session to be persistent) and most of people don't know that 0 is the hack to disable it.

Reply:
In reality it is the debug mode that is not working.
when you remove the handle_debug from handle_exception, you should understand easily the purpose of this one... It is to handle the debug params ... in 404 or 500 since the infos exposed differ according the debug mode.
Now you cannot enable/disable the debug mode when you are on a exception (404/500) page.
We rarely put code for pleasure, so when you rewrite and remove code from function, take a look to blame or try to guess why we need it :pleading_face: